tests/integration/repos_test.go

```diff
@@ -114,30 +114,36 @@ func TestRepositories_EditBranches(t *testing.T) {
 		t.Fatalf("Repositories.GetBranch() returned error: %v", err)
 	}
 
-	if *branch.Protection.Enabled {
+	if *branch.Protected {
 		t.Fatalf("Branch %v of repo %v is already protected", "master", *repo.Name)
 	}
 
-	branch.Protection.Enabled = github.Bool(true)
-	branch.Protection.RequiredStatusChecks = &github.RequiredStatusChecks{
-		EnforcementLevel: github.String("everyone"),
-		Contexts:         &[]string{"continous-integration"},
+	branch.Protected = github.Bool(true)
+
+	input := &github.ProtectionRequest{
+		RequiredStatusChecks: &github.RequiredStatusChecks{
+			IncludeAdmins: github.Bool(true),
+			Strict:        github.Bool(true),
+			Contexts:      &[]string{"continuous-integration"},
+		},
+		Restrictions: &github.BranchRestrictionsRequest{
+			Users: &[]string{"u"},
+			Teams: &[]string{"t"},
+		},
 	}
-	branch, _, err = client.Repositories.EditBranch(*repo.Owner.Login, *repo.Name, "master", branch)
+
+	protection, _, err := client.Repositories.UpdateBranchProtection("o", "r", "b", input)
 	if err != nil {
 		t.Fatalf("Repositories.EditBranch() returned error: %v", err)
 	}
 
-	if !*branch.Protection.Enabled {
+	if !*protection.RequiredStatusChecks.IncludeAdmins {
 		t.Fatalf("Branch %v of repo %v should be protected, but is not!", "master", *repo.Name)
```

Comment:
At the same time, is there anything that checks that branch is protected? Perhaps you should have a check like this:

```Go
if !*protection.Restrictions != ... {
    t.Fatalf("Branch %v of repo %v should be protected, but is not!", "master", *repo.Name)
}
```

I'm not exactly sure.